algorithm/CMakeLists.txt

```diff
@@ -9,6 +10,47 @@ target_link_libraries(
             vir
             magic_enum)
 
+set(GR_FILEIO_CPR_AVAILABLE FALSE)
+if(NOT EMSCRIPTEN)
+
+  if(NOT
+     GR_INSTALL_CPR
+     STREQUAL
+     "OFF")
+
+    find_package(CURL QUIET)
+
+    # ON -> always install CPR; OPTIONAL -> only install CPR if system libcurl is available
+    if(GR_INSTALL_CPR STREQUAL "ON" OR (GR_INSTALL_CPR STREQUAL "OPTIONAL" AND CURL_FOUND))
+
+      include(FetchContent)
+
+      if(CURL_FOUND)
+        set(CPR_USE_SYSTEM_CURL
+            ON
+            CACHE BOOL "Use system libcurl in CPR" FORCE)
+      endif()
+
+      FetchContent_Declare(
+        cpr
+        GIT_REPOSITORY https://github.com/libcpr/cpr.git
+        GIT_TAG 1.12.0)
+      FetchContent_MakeAvailable(cpr)
```

Comment:
The cmake documentation recommends to only call `FetchContent_MakeAvailable` once per project after all dependencies have been declared. This will work but if one of the dependencies declared in the top-level cmake will start using cpr at some point they might override our depencency.

For me it also would feel nicer to declare dependencies as close as possible to their usage, but probably this should be moved together with all the other dependencies.
https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable (paragraph above the code samples)